### PR TITLE
feat(shwap): shwap nodebuilder and cleanup

### PR DESCRIPTION
**Pull request by walldiss**
This PR:
- add integration of all shwap modules into nodebuilder 
- Full Pruner integration (2 lines)
- removes unused code
  - old storage related code and tests
  - old getter related code
- fixes linter errors
- fixes multiple tests
- fixes edsser util

There are still remaining unused code here and there that will be clean-up later. The goal of this PR is to make sure that
- build
- test 
- lint
- run 
Are running without errors

### CODE REVIEW

File: nodebuilder/p2p/bitswap.go [hunk 1]
```diff
@@ -26,10 +27,13 @@ const (
 	defaultBloomFilterHashes = 7
 	// default size of arc cache in blockStore
 	defaultARCCacheSize = 64 << 10
+	// TODO(@walldiss): expose cache size to cfg
+	// default blockstore cache size
+	defaultBlockstoreCacheSize = 128
```
Comment (walldiss) [inline nodebuilder/p2p/bitswap.go:32]: This will become part of config in different PR.

File: pruner/full/pruner.go [hunk 2]
```diff
@@ -4,36 +4,29 @@ import (
...
-	if eh.DAH.Equals(share.EmptyEDSRoots()) {
-		return nil
-	}
-
 	log.Debugf("pruning header %s", eh.DAH.Hash())
 
-	err := p.store.Remove(ctx, eh.DAH.Hash())
-	if err != nil && !errors.Is(err, dagstore.ErrShardUnknown) {
+	err := p.store.Remove(ctx, eh.Height(), eh.DAH.Hash())
+	if err != nil && !errors.Is(err, store.ErrNotFound) {
```
Comment (Wondertan) [inline pruner/full/pruner.go:29]: Store doesn't return this error. The method is now idempotent.
```suggestion
	err := p.store.Remove(ctx, eh.Height(), eh.DAH.Hash())
	if err != nil {
```

File: nodebuilder/tests/reconstruct_test.go [hunk 3]
```diff
@@ -1,5 +1,3 @@
-//go:build reconstruction || integration
```
Comment (Wondertan) [inline nodebuilder/tests/reconstruct_test.go:1]: Shouldn't this file contain skipping reconstruction tests?

File: nodebuilder/share/constructors.go [hunk 4]
```diff
@@ -1,57 +1,28 @@
...
-// upon request for a block with an empty data square.
-func ensureEmptyEDSInBS(ctx context.Context, bServ blockservice.BlockService) error {
-	_, err := ipld.AddShares(ctx, share.EmptyBlockShares(), bServ)
-	return err
-}
-
 func lightGetter(
 	shrexGetter *shrex_getter.Getter,
-	ipldGetter *getters.IPLDGetter,
+	bitswapGetter *bitswap.Getter,
```
Comment (Wondertan) [inline nodebuilder/share/constructors.go:18]: FN/BNs should similarly have bitswap getter.

File: nodebuilder/share/constructors.go [hunk 5]
```diff
@@ -60,11 +31,11 @@ func lightGetter(
 // by shrex the next time the data is retrieved (meaning shard recovery is
 // manual after corruption is detected).
 func bridgeGetter(
-	storeGetter *getters.StoreGetter,
+	storeGetter *store.Getter,
```
Comment (Wondertan) [inline nodebuilder/share/constructors.go]: Can unify getter consturtors now

File: share/shwap/p2p/bitswap/bitswap.go [hunk 6]
```diff
@@ -112,13 +118,35 @@ func NewServer(
...
+		bitswap.WithoutDuplicatedBlockStats(),
+
+		// server options
+		bitswap.ProvideEnabled(providesEnabled),
+		bitswap.SetSendDontHaves(sendDontHaves),
+		bitswap.MaxQueuedWantlistEntriesPerPeer(maxServerWantListsPerPeer),
+		bitswap.WithTargetMessageSize(targetMessageSize),
+		bitswap.MaxOutstandingBytesPerPeer(outstandingBytesPerPeer),
 	)
+
```
Comment (Wondertan) [inline share/shwap/p2p/bitswap/bitswap.go:148]: We can avoid duplicating this by defining 
```
type Bitswap {
 Client ...
 Server ...
}

func (b *Bitswap) NotifyNewBlocks() error {
  // calls both
}
```

We don't to define other methods and simply calling `network.Start(client, server)` should be enough to cover other methods.

File: share/shwap/p2p/bitswap/bitswap.go [hunk 7]
```diff
@@ -82,18 +83,23 @@ func NewClient(
 	ctx context.Context,
 	net network.BitSwapNetwork,
 	bstore blockstore.Blockstore,
+	opts ...client.Option,
```
Comment (Wondertan) [inline share/shwap/p2p/bitswap/bitswap.go:86]: Client doesn't need any options

File: share/shwap/p2p/bitswap/bitswap.go (hunk 6 above)
Comment (Wondertan) [reply share/shwap/p2p/bitswap/bitswap.go:148]: This also makes it easier for to reuse common Server accross two networks.

File: nodebuilder/p2p/bitswap.go [hunk 8]
```diff
@@ -36,10 +30,11 @@ func dataExchange(tp node.Type, params bitSwapParams) exchange.SessionExchange {
 	switch tp {
 	case node.Full, node.Bridge:
 		bs := bitswap.New(params.Ctx, net, params.Bs)
+		net.Start(bs.Client, bs.Server)
```
Comment (Wondertan) [inline nodebuilder/p2p/bitswap.go:33]: Did you try extracting bitswap construction to share pkg?  
(we can do that later as well)

File: share/shwap/p2p/shrex/shrexeds/server.go [hunk 9]
```diff
@@ -89,6 +89,7 @@ func (srv *Server) streamHandler(ctx context.Context) network.StreamHandler {
 func (s *Server) handleEDS(ctx context.Context, stream network.Stream) error {
 	logger := log.With("peer", stream.Conn().RemotePeer().String())
 	logger.Debug("server: handling eds request")
+	s.observeRateLimitedRequests()
```
Comment (Wondertan) [inline share/shwap/p2p/shrex/shrexeds/server.go:92]: TODO?

File: nodebuilder/share/bitswap.go [hunk 10]
```diff
@@ -0,0 +1,82 @@
...
+	"github.com/celestiaorg/celestia-node/store"
+)
+
+const (
+	// TODO(@walldiss): expose cache size to cfg
+	// default blockstore cache size
+	defaultBlockstoreCacheSize = 128
+)
+
+// dataExchange provides a constructor for IPFS block's DataExchange over BitSwap.
```
Comment (Wondertan) [inline nodebuilder/share/bitswap.go:26]:
```suggestion
// dataExchange constructs Exchange(Bitswap Composition) for Shwap
```

File: libs/edssser/edssser.go [hunk 11]
```diff
@@ -152,7 +140,7 @@ AvgTime %s
 	)
 }
 
-func (ss *EDSsser) put(ctx context.Context, t *testing.T) (time.Duration, error) {
+func (ss *EDSsser) put(ctx context.Context, t *testing.T, height int) (time.Duration, error) {
```
Comment (cristaloleg) [inline libs/edssser/edssser.go:143]: Why not `uint64` ? I know it's coming from `ss.config.WriteFrom` buuuut this sounds buggy due to cast to a smaller integer (in theory 😉 )

Comment (cristaloleg) [reply libs/edssser/edssser.go:143]: So bette to change `ss.config.WriteFrom` and `ss.config.EDSWrites` to `uint64`

File: das/daser_test.go [hunk 12]
```diff
@@ -438,3 +435,22 @@ func (m getterStub) GetRangeByHeight(
 func (m getterStub) Get(context.Context, libhead.Hash) (*header.ExtendedHeader, error) {
 	panic("implement me")
 }
+
+// waitHeight waits for the DASer to catch up to the given height. It will return an error if the
+// DASer fails to catch up to the given height within the timeout.
+func waitHeight(ctx context.Context, daser *DASer, height int) error {
```
Comment (cristaloleg) [inline das/daser_test.go:441]: Same here, `uint64`.

File: libs/edssser/edssser.go [hunk 13]
```diff
@@ -28,49 +27,38 @@ type Config struct {
...
-		return stats, err
-	}
-	fmt.Printf("recovered %d EDSes\n\n", len(edsHashes))
-
 	t := &testing.T{}
-	for toWrite := ss.config.EDSWrites - len(edsHashes); ctx.Err() == nil && toWrite > 0; toWrite-- {
-		took, err := ss.put(ctx, t)
+	writeTo := ss.config.WriteFrom + ss.config.EDSWrites
+	for height := ss.config.WriteFrom; ctx.Err() == nil && height < writeTo; height++ {
+		took, err := ss.put(ctx, t, height)
```
[the remark this answers is not in the document]
Comment (walldiss) [reply libs/edssser/edssser.go:61]: Looks like It is logged few lines down

File: libs/edssser/edssser.go (hunk 11 above)
Comment (walldiss) [reply libs/edssser/edssser.go:143]: The Edssser is minor side tool, that was written in low-effort and not worth a refactoring. I would rather avoid adding more changes to this already large PR

Comment (cristaloleg) [reply libs/edssser/edssser.go:143]: IMO on this scale changing `int` -> `uint64` is a small addition, `assert` -> `require` is already here.

**Review by Wondertan — APPROVED** [review]

**Comment by Wondertan** [comment]
This PR does a lot of things it shouldn't. It removes the old code and fixes lints. I want to express that this is an exception and we should not repeat that again.

**Pull request merged**